roadmap/implementors-guide/guide.md

```diff
@@ -989,12 +991,78 @@ Dispatch a `PovFetchSubsystemMessage(relay_parent, candidate_hash, sender)` and
 
 (TODO: send statements to Statement Distribution subsystem, handle shutdown signal from candidate backing subsystem)
 
+### Statement Gossip Subsystem
+
+The statement gossip subsystem is responsible for gossiping out and receiving [Signed Statements](#Signed-statement-type) from validators.
+
+#### Message Types
+
+This subsystem communicates with the overseer via [two message types](#Candidate-Backing-Subsystem-Messages), one for overseer -> subsystem and another for subsystem -> overseer. Currently this just differentiates between statements that the subsystem has been instructed to gossip out, and statements that the subsystem has received.
+
+#### Functionality
+
+The subsystem needs to contain a handle to a gossiping engine to gossip and recieve messages. This will be cloned into each job that is spawned. Apart from that, it also contains the general structures that all subsystems contain, e.g. channels to communicate with the overseer and handles to spawned jobs in order to shut them down.
```

Comment:
> This will be cloned into each job that is spawned.

Why clone the `GossipEngine` for each job? `GossipEngine` does not implement `Clone` by design, thus you would need to wrap it in an `Arc<Mutex<>>`. The statement gossip subsystem will run as a future, correct? If so wrapping a `GossipEngine` in a `Mutex` will prevent you from using `async/await` as (1) it is very difficult to make sure the lock is only held for short amount of time and definitely not while being preempted / sleeping and (2) a `MutexGuard` is not `Send` which would be required for `async/await`.

Does the above make sense? I am happy to expand further on this.

Reply:
Regardless, it seems reasonable for the gossip engine to have a single owner (the subsystem). A job for this subsystem would only entail stringing together `gossip_messages_for(specific_parent_hash_topic)` and sending a message to the other subsystem.